src/Xrefcheck/Verify.hs

```diff
@@ -160,9 +190,13 @@ verifyRepo
 
   progressRef <- newIORef $ initVerifyProgress (map snd toScan)
 
-  withAsync (printer progressRef) $ \_ ->
-    fmap fold . forConcurrently toScan $ \(file, ref) ->
-      verifyReference config mode progressRef repoInfo' root file ref
+  accumulated <- withAsync (printer progressRef) $ \_ ->
+    forConcurrentlyCaching toScan (\(_, Reference{..}) -> case locationType rLink of
+      ExternalLoc -> CacheUnderKey rLink
+      _           -> NoCaching) (\(file, ref) -> do
```

Comment:
Could you move this `case locationType rLink` to a separate (probably local) function? This way it should be simpler to read and update later.

And there will be no need for large parentheses.